### PR TITLE
API Documentation re-vamp (v.1) 

### LINKED ISSUE #6010: Support [index:string] as prop name in documentation parser
There is a typo on that page https://developer.microsoft.com/en-us/fabric#/components/link

image

### LINKED ISSUE #7309: Documentation not parsing entirely some enums
### Environment Information 
 
- __Package version(s)__: latest
- __Browser and OS versions__: all

### Please provide a reproduction of the bug in a codepen:

When running locally or directly on the demo website the documentation parser does not output all the enums values when they are strings as opposed to numbers. One good example is here:
https://github.com/OfficeDev/office-ui-fabric-react/blob/09f1011e7355358023bb8ec712defbe62f786475/packages/office-ui-fabric-react/src/components/HoverCard/HoverCard.types.ts#L137-L159
Here we have 2 enums. One is using numbers as values and the other one is using strings. The output of the documentation parser is this
screen shot 2018-12-05 at 11 41 20 am

#### Actual behavior:
 
The enums having strings as values in the documentation are displayed only partialy.
 
#### Expected behavior:
 
Should display all the values of the enum.

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? Maybe. Need to investigate how the parser works first. If there is anyone already knowing its internals probably they should handle it quicker.
 
Requested priority: Low
 
Products/sites affected: (if applicable)
Demo website https://developer.microsoft.com/en-us/fabric

### LINKED ISSUE #7819: Documentation Update - React UI Fabric Dropdown Control
Not really sure where I should file this but can selectedKey and defaultSelectedKey be added to the documentation for the React Dropdown Control:
https://developer.microsoft.com/en-us/fabric#/components/dropdown#Implementation
They seem to be missing from the Optional Members section despite having examples of it in the code; Lost a little bit of time on this.

Never filed one of these before so apologies if this is in the wrong area.

Thanks.

### PR DESCRIPTION
**Pull request by natalieethell**
#### Pull request checklist

- [X] Addresses an existing issue: Fixes a range of issues tracked in #6780: Fixes #7309, Fixes #6010, Fixes #7819   
- [x] Include a change request file using `$ npm run change`

#### Description of changes

**TL;DR:** Api documentation re-vamp that utilizes api-extractor output.

Thank you to everyone who took the time to look through the draft PR! All comments/feedback are incorporated in this new PR.

#### Design overview
These changes introduce a new package @uifabric/api-docs. This package takes the api.json files that api extractor generates and splits it up into individual page.json files to populate both component pages and (new) references pages. These page.json files live in @uifabric/api-docs.

The new references pages live under the left nav on both the Fabric site and the local dev site, as can be seen in the screenshots below.

Screen Shot 2019-04-15 at 5 03 34 PM

A sample reference page:

Screen Shot 2019-04-16 at 10 50 25 AM

Interface inheritance is now surfaced, there is hyperlinking between sections of a page and different pages, and deprecated properties are highlighted.

Screen Shot 2019-04-15 at 5 00 16 PM

Screen Shot 2019-04-25 at 1 34 11 PM

After discussing different implementations of hyperlinking, we decided that for now it's best to open a new tab when a link to another page is clicked. For example, when clicking on 'ITheme' that lives in a reference table here, a new tab will open for the 'ITheme' page. That way, a user's original page will stay open and they won't lose their place.

Screen Shot 2019-04-15 at 4 56 51 PM

The same page linking scrolling behavior from the original draft PR is maintained.

#### Future changes
There are several general improvements in mind for the next iteration of the api documentation clean-up/enhancement that Vitalie and I brainstormed:

* Fix deltas in navigation between the local/dev site and the Fabric site - this will work into the docs.microsoft.com work
* Document classes and types on component pages

#### Notes
The same page linking scrolling, for now, only works on the Fabric site, not on the local/dev site. If you want to see those animations, feel free to checkout my branch, build it locally, and run "npm start" in apps/fabric-website.

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8844)

### CODE REVIEW

File: apps/fabric-website/src/components/App/App.tsx [hunk 1]
```diff
@@ -58,7 +59,11 @@ export class App extends React.Component<IAppProps, any> {
 
   public componentWillReceiveProps(nextProps: IAppProps): void {
     if (nextProps && nextProps.children !== this.props.children) {
-      document.body.scrollTop = document.documentElement.scrollTop = 0;
+      const hash = extractAnchorLink(window.location.hash);
+      // when loading a new component page, we want to scroll to the top
+      if (hash.indexOf('/components') !== -1) {
+        document.body.scrollTop = document.documentElement.scrollTop = 0;
```
Comment (KevinTCoughlin) [inline apps/fabric-website/src/components/App/App.tsx]: I don't think you need both of these assignments. You might consider `window.scrollTo(0, 0)` via [Window.scrollTo](https://developer.mozilla.org/en-US/docs/Web/API/Window/scrollTo) or just `document.documentElement.scrollTop` which is `html` DOMNode rather than `body` since the intent is to scroll the entire page.

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx [hunk 2]
```diff
@@ -0,0 +1,511 @@
...
+        isResizable: true,
+        onRender: this.createRenderCellEnum('description')
+      }
+    ];
+  }
+
+  public renderCell = (text: string, deprecated: boolean = false) => {
+    // When the text is passed to this function, it has had newline characters removed,
+    // so this regex will match backtick sequences that span multiple lines.
+    const regex = new RegExp('`[^`]*`', 'g');
```
Comment (KevinTCoughlin) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: Consider hoisting and defining this `RegExp` once since it is currently called in a loop for rendering.

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+            undefined
+          )}
+          <Text block variant="small">
+            {text}
+          </Text>
+        </>
+      );
+    }
+
+    const eltChildren = this._extractCodeBlocks(text, codeBlocks);
```
Comment (KevinTCoughlin) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: You might consider renaming this `eltChildren` variable for future maintainers. I wasn't sure what it meant. Or provide a descriptive comment, perhaps its just me.

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx [hunk 4]
```diff
@@ -0,0 +1,511 @@
...
+        </Text>
+      </>
+    );
+  };
+
+  public renderCellType = (typeTokens: ILinkToken[]) => {
+    return this._parseILinkTokens(false, typeTokens);
+  };
+
+  public createRenderCellEnum = (propertyName: keyof IApiEnumProperty) => (item: IApiEnumProperty) => this.renderCell(item[propertyName]);
```
Comment (KevinTCoughlin) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: Do all these methods need to be public? If not consider hiding their visibility via `private`. If they do, feel free to resolve.

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx [hunk 5]
```diff
@@ -0,0 +1,511 @@
...
+            <br />
+          </>
+        ) : (
+          undefined
+        )}
+        <Text block variant="small">
+          {eltChildren}
+        </Text>
+      </>
+    );
```
Comment (KevinTCoughlin) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: It seems like this component could be shared as a functional component so that its not defined more than once.

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx [hunk 6]
```diff
@@ -0,0 +1,511 @@
...
+  }
+
+  private _onRenderRow = (props: IDetailsRowProps, defaultRender?: IRenderFunction<IDetailsRowProps>): JSX.Element => {
+    const { item } = props;
+    const rowStyles: Partial<IDetailsRowStyles> = {
+      root: {
+        selectors: {
+          ':hover': {
+            background: 'transparent'
```
Comment (KevinTCoughlin) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]:
```suggestion
            background: 'none'
```

Curious, does the above not work?

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx [hunk 7]
```diff
@@ -0,0 +1,511 @@
...
+          background: DEPRECATED_ROW_COLOR,
+          selectors: {
+            ':hover': {
+              background: DEPRECATED_ROW_COLOR
+            }
+          }
+        }
+      };
+
+      return <DetailsRow {...props} styles={rowStyles && deprecatedStyles} />;
```
Comment (KevinTCoughlin) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]:
```suggestion
      return <DetailsRow {...props} styles={{ ...rowStyles, ...deprecatedStyles }} />;
```

Assuming you want to merge the styles I think the above would be more clear.

**OR** you can probably just do this merge _once_ and use it in this conditional block.

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx [hunk 8]
```diff
@@ -0,0 +1,511 @@
...
+
+  private _parseILinkTokensHelper = (linkTokens: ILinkToken[], extend: boolean): JSX.Element | undefined => {
+    return (
+      <>
+        {linkTokens.map((token: ILinkToken, index: number) => {
+          let match: RegExpMatchArray | null = null;
+          let hash: string = '/components/';
+          // get hash to set correct href value on the links for the local site vs. the Fabric site
+          if (this._baseUrl !== '') {
+            match = this._baseUrl.match(/\/\w+\//);
```
Comment (KevinTCoughlin) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: Nit, might be able to create this RegExp once and re-use.

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx [hunk 9]
```diff
@@ -0,0 +1,265 @@
...
+  }
+
+  public componentDidUpdate(prevProps: IApiReferencesTableSetProps, prevState: IApiReferencesTableSetState): void {
+    if (prevState.showSeeMore === false && this.state.showSeeMore === true) {
+      const hash = extractAnchorLink(window.location.hash);
+      const el = document.getElementById(hash);
+
+      if (el) {
+        // update scroll position
+        window.scrollTo({ top: el.offsetTop - TITLE_LINE_HEIGHT /* title */, behavior: 'smooth' });
```
Comment (KevinTCoughlin) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx]: Shame you need the offset here otherwise [Element.scrollIntoView](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollIntoView) would have worked.

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx [hunk 10]
```diff
@@ -0,0 +1,265 @@
...
+
+  private _renderEach(): JSX.Element | undefined {
+    if (this.state.properties.length > 1) {
+      return (
+        <Stack gap={MEDIUM_GAP_SIZE}>
+          <ActionButton
+            iconProps={{ iconName: this.state.showSeeMore ? 'SkypeCircleMinus' : 'CirclePlus' }}
+            onClick={this._onClickSeeMore}
+            onRenderText={this._onRenderText}
+            styles={{ root: { paddingLeft: '0px' }, textContainer: { paddingLeft: '4px' } }}
```
Comment (KevinTCoughlin) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx]: Not that perf is a primary concern for documentation, but _I think_ in-lining these new Objects causes React to re-reconcile vs. if the objects were stored in some way.

Since these are likely only rendered once per page load, the above concern doesn't matter.

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx (hunk 9 above)
Comment (KevinTCoughlin) [reply packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx]: You might consider sharing this function tho

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx [hunk 11]
```diff
@@ -0,0 +1,265 @@
...
+      property: interfaceMembers
+    };
+  }
+
+  private _generateClassProperty(table: ITableJson): IApiProperty {
+    // class members are a mix of IApiInterfaceProperty and IMethod
+    const classMembers: IApiInterfaceProperty[] = [];
+    const classMethods: IMethod[] = [];
+
+    const members: ITableRowJson[] = table.members as ITableRowJson[];
```
Comment (KevinTCoughlin) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx]: These casts can probably be removed with some refactoring, but we can do that separately.

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx (hunk 3 above)
Comment (natalieethell) [reply packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: Good call!

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx (hunk 6 above)
Comment (natalieethell) [reply packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: Nope, it definitely works, just honestly didn't think to do that!

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx (hunk 4 above)
Comment (natalieethell) [reply packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: Nope not at all!

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx (hunk 11 above)
Comment (natalieethell) [reply packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx]: Will keep this in mind for the next PR!

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx (hunk 9 above)
Comment (natalieethell) [reply packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx]: I'm planning on improving the scrolling in a next iteration - I'll keep this in mind :)

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx [hunk 12]
```diff
@@ -0,0 +1,512 @@
...
+  methods?: IMethod[];
+  renderAsEnum?: boolean;
+  renderAsClass?: boolean;
+  key?: string;
+  name?: string;
+  description?: string;
+  extendsTokens?: ILinkToken[];
+}
+
+export interface IApiReferencesTableState {
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: This component shouldn't use state, because you never update the state. Instead, just store the relevant things in private members (and extend React.PureComponent rather than React.Component).

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx [hunk 13]
```diff
@@ -0,0 +1,512 @@
...
+  private _defaultColumns: IColumn[];
+  private _methodColumns: IColumn[];
+  private _enumColumns: IColumn[];
+
+  constructor(props: IApiReferencesTableProps) {
+    super(props);
+
+    if (props.renderAsEnum) {
+      const properties = (props.properties as IApiEnumProperty[])
+        .sort((a: IApiEnumProperty, b: IApiEnumProperty) => (a.value < b.value ? -1 : a.value > b.value ? 1 : 0))
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: Would be nice to put the duplicated sorting code in a helper.

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx [hunk 14]
```diff
@@ -0,0 +1,512 @@
...
+  public render(): JSX.Element | null {
+    const { description, extendsTokens } = this.props;
+    const { properties, isEnum, isClass } = this.state;
+
+    return (
+      <Stack gap={MEDIUM_GAP_SIZE}>
+        {description && extendsTokens && extendsTokens.length > 0 ? (
+          <Stack gap={SMALL_GAP_SIZE}>
+            {this._renderTitle()}
+            {(description || (extendsTokens && extendsTokens.length > 0)) && (
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: This check is redundant since you already checked at the beginning of this block that both description and extendsTokens exist

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx [hunk 15]
```diff
@@ -0,0 +1,512 @@
...
+                items={properties}
+                columns={isEnum ? this._enumColumns : this._defaultColumns}
+                onRenderRow={this._onRenderRow}
+              />
+            )}
+      </Stack>
+    );
+  }
+
+  private renderCell = (text: string, deprecated: boolean = false) => {
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: nit, private methods are typically prefixed with _ (we should probably consider a lint rule enforcing this)

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx [hunk 16]
```diff
@@ -0,0 +1,512 @@
...
+    }
+
+    return textElements;
+  }
+
+  private _renderClass(): JSX.Element | undefined {
+    const { properties, methods } = this.state;
+
+    return methods ? (
+      <Stack gap={MEDIUM_GAP_SIZE}>
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: Not sure nested stacks are the best way to go here, since there are only two elements per stack, and stacks use flexbox (which does have a higher perf cost). Might be better to just use margin after the text or before the list instead. (Also, would be nice if the "Members" part could be de-duplicated.)

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx [hunk 17]
```diff
@@ -0,0 +1,512 @@
...
+          background: DEPRECATED_ROW_COLOR,
+          selectors: {
+            ':hover': {
+              background: DEPRECATED_ROW_COLOR
+            }
+          }
+        }
+      };
+
+      return <DetailsRow {...props} styles={{ ...rowStyles, ...deprecatedStyles }} />;
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: Is the intend here that root on the deprecated style entirely overwrite root on the non-deprecated version?

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx [hunk 18]
```diff
@@ -0,0 +1,512 @@
...
+              hash = match[0];
+            }
+          }
+          if (token.pageKind && token.hyperlinkedPage) {
+            let href;
+            const split = this._baseUrl.split('#');
+            const cleanedSplit = split.filter(value => !!value);
+
+            // whether the link should be opened in a new tab, defaults to true
+            let newTab = true;
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: We should look at similar doc sites to see if their links open in a new tab

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx [hunk 19]
```diff
@@ -0,0 +1,512 @@
...
+
+            if (token.pageKind === 'References') {
+              const referencePage = hash + token.pageKind.toLowerCase() + '/' + token.hyperlinkedPage.toLowerCase();
+              href = '#' + referencePage + '#' + token.text;
+              if (cleanedSplit.length > 1) {
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: nit: use && not nested ifs

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx [hunk 20]
```diff
@@ -0,0 +1,281 @@
...
+import { mergeStyles } from 'office-ui-fabric-react/lib/Styling';
+import { css } from 'office-ui-fabric-react/lib/Utilities';
+import { ApiReferencesTable, MEDIUM_GAP_SIZE, LARGE_GAP_SIZE } from './ApiReferencesTable';
+import { IApiProperty, IApiInterfaceProperty, IApiEnumProperty, IMethod, IApiReferencesTableSetProps } from './ApiReferencesTableSet.types';
+import { IEnumTableRowJson, ITableRowJson, ITableJson } from 'office-ui-fabric-react/lib/common/DocPage.types';
+import { PropertyType } from '../../utilities/parser/index';
+import { extractAnchorLink } from '../../utilities/extractAnchorLink';
+
+export interface IApiReferencesTableSetState {
+  properties: Array<IApiProperty>;
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx]: These should be in a private member, not the state. State is only for things that can change during the life of the component (and ideally only changes from things like user interaction or server data fetching, not props updates).

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx [hunk 21]
```diff
@@ -0,0 +1,281 @@
...
+    }
+
+    return [...iComponentProps, ...preResults];
+  }
+
+  private _generateEnumProperty(table: ITableJson): IApiProperty {
+    const enumMembers: IApiEnumProperty[] = [];
+
+    const members: IEnumTableRowJson[] = table.members as IEnumTableRowJson[];
+    for (let k = 0; k < members.length; k++) {
```
Comment (ecraig12345) [inline packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTableSet.tsx]: nit, I suggest using `map` here (and any time there's a 1-1 mapping from items of the original array to a new array)

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx (hunk 16 above)
Comment (ecraig12345) [reply packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: @khmakoto Do you know how much of a performance concern it is to have nested stacks like this? (I vaguely remember it being a concern, but I don't remember the details)

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx (hunk 18 above)
Comment (natalieethell) [reply packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: This was design discussed with @aditima @Vitalius1 @KevinTCoughlin @aftab-hassan @margoflynn and @iclanton. If we're not a fan of this, we can iterate off of it, but we thought this would be the best approach for now.

**Comment by cliffkoh** [comment]
@natalieethell @ecraig12345  is it just me or do the links not work at http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/8844/merge/fabric-website-resources/dist/index.html#/examples/datepicker

**Comment by natalieethell** [comment]
@cliffkoh the url structure differs for the PR deploy site, which I didn't take into account, which is probably why those aren't working

**Comment by natalieethell** [comment]
@cliffkoh let me take a closer look, though, to ensure that those links aren't affected in the actual fabric site

**Comment by cliffkoh** [comment]
Will it work for aka.ms/fabricdemo site?

**Comment by natalieethell** [comment]
@cliffkoh I'll probably have to make a change for that scenario, working on it now

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx (hunk 16 above)
Comment (khmakoto) [reply packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: Using flexbox in and of itself introduces performance issues. I don't have a specific metric on how much its use impacts performance. I would say that for these two item scenarios it might be better to just have a margin on the `Text`, especially if the `Stacks` are only being used to add a gap between the items. However, if the Stacks are being used for other layout purposes, then I would say using nested `Stacks` might still be a good call.

Comment (KevinTCoughlin) [reply packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: Let's revisit this with snapshot captures / measurements given the scenarios we need to support.

File: packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx (hunk 18 above)
Comment (KevinTCoughlin) [reply packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: The justification for the "New tab" experience when clicking a hyper-link was so that folks reading API references did not lose their place (scroll position and URL) when looking up related API references. Let's see how it works in practice and revise if need be.

Comment (ecraig12345) [reply packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: I can kind of see that perspective, but personally I find it incredibly annoying if I *didn't* want a new tab, because then I have to go back and close the old one--whereas if I *do* want a new tab, I can easily make that happen with ctrl/cmd+click. For scroll position, natively I think browsers tend to do a pretty good job remembering it these days, but that might be negated by our custom scroll handling.

Also, FWIW, the handful of the doc sites I looked at just now **don't** use new tabs: [Node docs](https://nodejs.org/api/crypto.html), [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference), [.NET docs](https://docs.microsoft.com/en-us/dotnet/) (and other stuff on docs.microsoft.com), [Java docs](https://docs.oracle.com/javase/10/docs/api/overview-summary.html).

Comment (cliffkoh) [reply packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: I agree with @ecraig12345 here. Most ideal is some kind of "peek" but navigating to the linked item is far more aligned with what I usually encounter, including in various IDEs (they don't open the code in a new tab when you navigate to definition).

Comment (KevinTCoughlin) [reply packages/example-app-base/src/components/ApiReferencesTable/ApiReferencesTable.tsx]: IIRC our custom scroll behavior was the problem. You can see this today on our documentation site where you are reset to `top: 0`. We felt that given the value of the information we are now surfacing, that the trade-off was acceptable in the short-term.

@natalieethell, let's follow-up and see if we can fix the root cause. Then we can relax enforcing new tab behavior, which I think we're all in favor of if scroll position is restored.
